<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Processing quantitative proteomics data with QFeatures • QFeatures</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Processing quantitative proteomics data with QFeatures">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QFeatures</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.19.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/QFeatures.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Processing.html">Processing quantitative proteomics data with QFeatures</a></li>
    <li><a class="dropdown-item" href="../articles/read_QFeatures.html">Load mass spectrometry-based proteomics data using `readQFeatures()`</a></li>
    <li><a class="dropdown-item" href="../articles/Visualization.html">Data visualization from a QFeatures object</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/RforMassSpectrometry/QFeatures/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Processing quantitative proteomics data with QFeatures</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/RforMassSpectrometry/QFeatures/blob/master/vignettes/Processing.Rmd" class="external-link"><code>vignettes/Processing.Rmd</code></a></small>
      <div class="d-none name"><code>Processing.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to process quantitative mass
      spectrometry data with QFeatures: cleaning up unneeded feature
      variables, adding an experimental design, filtering out
      contaminants and reverse hits, managing missing values,
      log-transforming, normalising and aggregating data. This vignette
      is distributed under a CC BY-SA license.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="reading-data-as-qfeatures">Reading data as <code>QFeatures</code><a class="anchor" aria-label="anchor" href="#reading-data-as-qfeatures"></a>
</h2>
<p>We are going to use a subset of the CPTAC study 6 containing
conditions A and B <span class="citation">(Paulovich et al.
2010)</span>. The peptide-level data, as processed by MaxQuant <span class="citation">(Cox and Mann 2008)</span> is available in the
<code>msdata</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu">msdata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msdata/man/proteomics.html" class="external-link">quant</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"cptac"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cptac_a_b_peptides.txt"</span></span></code></pre>
<p>From the names of the columns, we see that the quantitative columns,
starting with <code>"Intensity."</code> (note the dot!) are at positions
56 to 61.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "Sequence"                 "N.term.cleavage.window"  </span></span>
<span><span class="co">##  [3] "C.term.cleavage.window"   "Amino.acid.before"       </span></span>
<span><span class="co">##  [5] "First.amino.acid"         "Second.amino.acid"       </span></span>
<span><span class="co">##  [7] "Second.last.amino.acid"   "Last.amino.acid"         </span></span>
<span><span class="co">##  [9] "Amino.acid.after"         "A.Count"                 </span></span>
<span><span class="co">## [11] "R.Count"                  "N.Count"                 </span></span>
<span><span class="co">## [13] "D.Count"                  "C.Count"                 </span></span>
<span><span class="co">## [15] "Q.Count"                  "E.Count"                 </span></span>
<span><span class="co">## [17] "G.Count"                  "H.Count"                 </span></span>
<span><span class="co">## [19] "I.Count"                  "L.Count"                 </span></span>
<span><span class="co">## [21] "K.Count"                  "M.Count"                 </span></span>
<span><span class="co">## [23] "F.Count"                  "P.Count"                 </span></span>
<span><span class="co">## [25] "S.Count"                  "T.Count"                 </span></span>
<span><span class="co">## [27] "W.Count"                  "Y.Count"                 </span></span>
<span><span class="co">## [29] "V.Count"                  "U.Count"                 </span></span>
<span><span class="co">## [31] "Length"                   "Missed.cleavages"        </span></span>
<span><span class="co">## [33] "Mass"                     "Proteins"                </span></span>
<span><span class="co">## [35] "Leading.razor.protein"    "Start.position"          </span></span>
<span><span class="co">## [37] "End.position"             "Unique..Groups."         </span></span>
<span><span class="co">## [39] "Unique..Proteins."        "Charges"                 </span></span>
<span><span class="co">## [41] "PEP"                      "Score"                   </span></span>
<span><span class="co">## [43] "Identification.type.6A_7" "Identification.type.6A_8"</span></span>
<span><span class="co">## [45] "Identification.type.6A_9" "Identification.type.6B_7"</span></span>
<span><span class="co">## [47] "Identification.type.6B_8" "Identification.type.6B_9"</span></span>
<span><span class="co">## [49] "Experiment.6A_7"          "Experiment.6A_8"         </span></span>
<span><span class="co">## [51] "Experiment.6A_9"          "Experiment.6B_7"         </span></span>
<span><span class="co">## [53] "Experiment.6B_8"          "Experiment.6B_9"         </span></span>
<span><span class="co">## [55] "Intensity"                "Intensity.6A_7"          </span></span>
<span><span class="co">## [57] "Intensity.6A_8"           "Intensity.6A_9"          </span></span>
<span><span class="co">## [59] "Intensity.6B_7"           "Intensity.6B_8"          </span></span>
<span><span class="co">## [61] "Intensity.6B_9"           "Reverse"                 </span></span>
<span><span class="co">## [63] "Potential.contaminant"    "id"                      </span></span>
<span><span class="co">## [65] "Protein.group.IDs"        "Mod..peptide.IDs"        </span></span>
<span><span class="co">## [67] "Evidence.IDs"             "MS.MS.IDs"               </span></span>
<span><span class="co">## [69] "Best.MS.MS"               "Oxidation..M..site.IDs"  </span></span>
<span><span class="co">## [71] "MS.MS.Count"</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Intensity\\."</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 56 57 58 59 60 61</span></span></code></pre>
<p>We now read these data using the <code>readQFeatures</code> function.
The peptide level expression data will be imported into R as an instance
of class <code>QFeatures</code> named <code>cptac</code> with an assay
named <code>peptides</code>. We also use the <code>fnames</code>
argument to set the row-names of the <code>peptides</code> assay to the
peptide sequences.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/RforMassSpectrometry/QFeatures" class="external-link">"QFeatures"</a></span><span class="op">)</span></span>
<span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readQFeatures.html">readQFeatures</a></span><span class="op">(</span><span class="va">x</span>, quantCols <span class="op">=</span> <span class="va">i</span>, name <span class="op">=</span> <span class="st">"peptides"</span>, fnames <span class="op">=</span> <span class="st">"Sequence"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Checking arguments.</span></span></code></pre>
<pre><code><span><span class="co">## Loading data as a 'SummarizedExperiment' object.</span></span></code></pre>
<pre><code><span><span class="co">## Formatting sample annotations (colData).</span></span></code></pre>
<pre><code><span><span class="co">## Formatting data as a 'QFeatures' object.</span></span></code></pre>
<pre><code><span><span class="co">## Setting assay rownames.</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 1 set:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 11466 rows and 6 columns</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="encoding-the-experimental-design">Encoding the experimental design<a class="anchor" aria-label="anchor" href="#encoding-the-experimental-design"></a>
</h2>
<p>Below we update the sample (column) annotations to encode the two
groups, 6A and 6B, and the original sample numbers.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6A"</span>, <span class="st">"6B"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">cptac</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">7</span><span class="op">:</span><span class="fl">9</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 2 columns</span></span>
<span><span class="co">##                      group    sample</span></span>
<span><span class="co">##                &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Intensity.6A_7          6A         7</span></span>
<span><span class="co">## Intensity.6A_8          6A         8</span></span>
<span><span class="co">## Intensity.6A_9          6A         9</span></span>
<span><span class="co">## Intensity.6B_7          6B         7</span></span>
<span><span class="co">## Intensity.6B_8          6B         8</span></span>
<span><span class="co">## Intensity.6B_9          6B         9</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="filtering-out-contaminants-and-reverse-hits">Filtering out contaminants and reverse hits<a class="anchor" aria-label="anchor" href="#filtering-out-contaminants-and-reverse-hits"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">cptac</span>, <span class="op">~</span> <span class="va">Reverse</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'Reverse' found in 1 out of 1 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 1 set:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 11436 rows and 6 columns</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">cptac</span>, <span class="op">~</span> <span class="va">Potential.contaminant</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'Potential.contaminant' found in 1 out of 1 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 1 set:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 11385 rows and 6 columns</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="va">cptac</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Reverse</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Potential.contaminant</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'Reverse' found in 1 out of 1 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## 'Potential.contaminant' found in 1 out of 1 assay(s).</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="removing-up-unneeded-feature-variables">Removing up unneeded feature variables<a class="anchor" aria-label="anchor" href="#removing-up-unneeded-feature-variables"></a>
</h2>
<p>The spreadsheet that was read above contained numerous variables that
are returned by MaxQuant, but not necessarily necessary in the frame of
a downstream statistical analysis.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-class.html">rowDataNames</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 1</span></span>
<span><span class="co">## [["peptides"]] Sequence N.term.cleavage.window ... MS.MS.Count</span></span></code></pre>
<p>The only ones that we will be needing below are the peptides
sequences and the protein identifiers. Below, we store these variables
of interest and filter them using the <code>selectRowData</code>
function.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rowvars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sequence"</span>, <span class="st">"Proteins"</span>, <span class="st">"Leading.razor.protein"</span><span class="op">)</span></span>
<span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-class.html">selectRowData</a></span><span class="op">(</span><span class="va">cptac</span>, <span class="va">rowvars</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/QFeatures-class.html">rowDataNames</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 1</span></span>
<span><span class="co">## [["peptides"]] Sequence Proteins Leading.razor.protein</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="managing-missing-values">Managing missing values<a class="anchor" aria-label="anchor" href="#managing-missing-values"></a>
</h2>
<p>Missing values can be very numerous in certain proteomics experiments
and need to be dealt with carefully. The first step is to assess their
presence across samples and features. But before being able to do so, we
need to replace 0 by <code>NA</code>, given that MaxQuant encodes
missing data with a 0 using the <code>zeroIsNA</code> function.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-missing-data.html">zeroIsNA</a></span><span class="op">(</span><span class="va">cptac</span>, i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/QFeatures-missing-data.html">nNA</a></span><span class="op">(</span><span class="va">cptac</span>, i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $nNA</span></span>
<span><span class="co">## DataFrame with 1 row and 3 columns</span></span>
<span><span class="co">##         assay       nNA       pNA</span></span>
<span><span class="co">##   &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1    peptides     30609  0.449194</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $nNArows</span></span>
<span><span class="co">## DataFrame with 11357 rows and 4 columns</span></span>
<span><span class="co">##             assay          name       nNA       pNA</span></span>
<span><span class="co">##       &lt;character&gt;   &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1        peptides AAAAGAGGAG...         4  0.666667</span></span>
<span><span class="co">## 2        peptides     AAAALAGGK         0  0.000000</span></span>
<span><span class="co">## 3        peptides    AAAALAGGKK         0  0.000000</span></span>
<span><span class="co">## 4        peptides AAADALSDLE...         0  0.000000</span></span>
<span><span class="co">## 5        peptides AAADALSDLE...         0  0.000000</span></span>
<span><span class="co">## ...           ...           ...       ...       ...</span></span>
<span><span class="co">## 11353    peptides YYSIYDLGNN...         6  1.000000</span></span>
<span><span class="co">## 11354    peptides YYTFNGPNYN...         3  0.500000</span></span>
<span><span class="co">## 11355    peptides    YYTITEVATR         4  0.666667</span></span>
<span><span class="co">## 11356    peptides YYTVFDRDNN...         6  1.000000</span></span>
<span><span class="co">## 11357    peptides YYTVFDRDNN...         6  1.000000</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $nNAcols</span></span>
<span><span class="co">## DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">##         assay          name       nNA       pNA</span></span>
<span><span class="co">##   &lt;character&gt;   &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1    peptides Intensity....      4669  0.411112</span></span>
<span><span class="co">## 2    peptides Intensity....      5388  0.474421</span></span>
<span><span class="co">## 3    peptides Intensity....      5224  0.459981</span></span>
<span><span class="co">## 4    peptides Intensity....      4651  0.409527</span></span>
<span><span class="co">## 5    peptides Intensity....      5470  0.481641</span></span>
<span><span class="co">## 6    peptides Intensity....      5207  0.458484</span></span></code></pre>
<p>The output of the <code>nNA</code> function tells us that</p>
<ul>
<li>there are currently close to 50% is missing values in the data;</li>
<li>there are 4051 peptides with 0 missing values, 989 with a single
missing values, … and 3014 peptides composed of only missing
values;</li>
<li>the range of missing values in the 6 samples is comparable and
ranges between 4651 and 5470.</li>
</ul>
<p>In this dataset, we have such a high number of peptides without any
data because the 6 samples are a subset of a larger dataset, and these
peptides happened to be absent in groups A and B. Below, we use
<code>filterNA</code> to remove all the peptides that contain one or
more missing values by using <code>pNA = 0</code> (which also is the
default value).</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-missing-data.html">filterNA</a></span><span class="op">(</span><span class="va">cptac</span>, i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">cptac</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 1 set:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 4051 rows and 6 columns</span></span></code></pre>
<p>I we wanted to keep peptides that have up to 90% of missing values,
corresponsing in this case to those that have only one value (i.e 5/6
percent of missing values), we could have set <code>pNA</code> to
0.9.</p>
</div>
<div class="section level2">
<h2 id="counting-unique-features">Counting unique features<a class="anchor" aria-label="anchor" href="#counting-unique-features"></a>
</h2>
<p>Counting the number of unique features across samples can be used for
quality control or for assessing the identification efficiency between
different conditions or experimental set-ups.
<code>countUniqueFeatures</code> can be used to count the number of
features that are contained in each sample of an assay from a
<code>QFeatures</code> object. For instance, we can count the number of
(non-missing) peptides per sample from the <code>peptides</code> assay.
Note that the counts are automatically stored in the
<code>colData</code> of <code>cptac</code>, under
<code>peptide_counts</code>:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/countUniqueFeatures.html">countUniqueFeatures</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>                             i <span class="op">=</span> <span class="st">"peptides"</span>,</span>
<span>                             colDataName <span class="op">=</span> <span class="st">"peptide_counts"</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">##                      group    sample peptide_counts</span></span>
<span><span class="co">##                &lt;character&gt; &lt;integer&gt;      &lt;integer&gt;</span></span>
<span><span class="co">## Intensity.6A_7          6A         7           4051</span></span>
<span><span class="co">## Intensity.6A_8          6A         8           4051</span></span>
<span><span class="co">## Intensity.6A_9          6A         9           4051</span></span>
<span><span class="co">## Intensity.6B_7          6B         7           4051</span></span>
<span><span class="co">## Intensity.6B_8          6B         8           4051</span></span>
<span><span class="co">## Intensity.6B_9          6B         9           4051</span></span></code></pre>
<p>We can also count the number of unique proteins. We therefore need to
tell <code>countUniqueFeatures</code> that we need to group by protein
(the protein name is stored in the <code>rowData</code> under
<code>Proteins</code>):</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/countUniqueFeatures.html">countUniqueFeatures</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>                             i <span class="op">=</span> <span class="st">"peptides"</span>,</span>
<span>                             groupBy <span class="op">=</span> <span class="st">"Proteins"</span>,</span>
<span>                             colDataName <span class="op">=</span> <span class="st">"protein_counts"</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">cptac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">##                      group    sample peptide_counts protein_counts</span></span>
<span><span class="co">##                &lt;character&gt; &lt;integer&gt;      &lt;integer&gt;      &lt;integer&gt;</span></span>
<span><span class="co">## Intensity.6A_7          6A         7           4051           1125</span></span>
<span><span class="co">## Intensity.6A_8          6A         8           4051           1125</span></span>
<span><span class="co">## Intensity.6A_9          6A         9           4051           1125</span></span>
<span><span class="co">## Intensity.6B_7          6B         7           4051           1125</span></span>
<span><span class="co">## Intensity.6B_8          6B         8           4051           1125</span></span>
<span><span class="co">## Intensity.6B_9          6B         9           4051           1125</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="imputation">Imputation<a class="anchor" aria-label="anchor" href="#imputation"></a>
</h2>
<p>The <code>impute</code> method can be used to perform missing value
imputation using a variety of imputation methods. The method takes an
instance of class <code>QFeatures</code> (or a
<code>SummarizedExperiment</code>) as input, an a character naming the
desired method (see <code><a href="../reference/impute.html">?impute</a></code> for the complete list with
details) and returns a new instance of class <code>QFeatures</code> (or
<code>SummarizedExperiment</code>) with imputed data.</p>
<p>As described in more details in <span class="citation">(Lazar et al.
2016)</span>, there are two types of mechanisms resulting in missing
values in LC/MSMS experiments.</p>
<ul>
<li><p>Missing values resulting from absence of detection of a feature,
despite ions being present at detectable concentrations. For example in
the case of ion suppression or as a result from the stochastic,
data-dependent nature of the MS acquisition method. These missing value
are expected to be randomly distributed in the data and are defined as
<em>missing at random</em> (MAR) or <em>missing completely at
random</em> (MCAR).</p></li>
<li><p>Biologically relevant missing values, resulting from the
<em>absence</em> of the low abundance of ions (below the limit of
detection of the instrument). These missing values are not expected to
be randomly distributed in the data and are defined as <em>missing not
at random</em> (MNAR).</p></li>
</ul>
<p>MAR and MCAR values can be reasonably well tackled by many imputation
methods. MNAR data, however, requires some knowledge about the
underlying mechanism that generates the missing data, to be able to
attempt data imputation. MNAR features should ideally be imputed with a
<em>left-censor</em> (for example using a deterministic or probabilistic
minimum value) method. Conversely, it is recommended to use <em>hot
deck</em> methods (for example nearest neighbour, maximum likelihood,
etc) when data are missing at random.</p>
<div class="figure">
<img src="Processing_files/figure-html/miximp-1.png" alt="Mixed imputation method. Black cells represent presence of quantitation values and light grey corresponds to missing data. The two groups of interest are depicted in green and blue along the heatmap columns. Two classes of proteins are annotated on the left: yellow are proteins with randomly occurring missing values (if any) while proteins in brown are candidates for non-random missing value imputation." width="700"><p class="caption">
Mixed imputation method. Black cells represent presence of quantitation
values and light grey corresponds to missing data. The two groups of
interest are depicted in green and blue along the heatmap columns. Two
classes of proteins are annotated on the left: yellow are proteins with
randomly occurring missing values (if any) while proteins in brown are
candidates for non-random missing value imputation.
</p>
</div>
<p>It is anticipated that the identification of both classes of missing
values will depend on various factors, such as feature intensities and
experimental design. Below, we use perform mixed imputation, applying
nearest neighbour imputation on the 654 features that are assumed to
contain randomly distributed missing values (if any) (yellow on figure
@ref(fig:miximp)) and a deterministic minimum value imputation on the 35
proteins that display a non-random pattern of missing values (brown on
figure @ref(fig:miximp)).</p>
</div>
<div class="section level2">
<h2 id="data-transformation">Data transformation<a class="anchor" aria-label="anchor" href="#data-transformation"></a>
</h2>
<p>When analysing continuous data using parametric methods (such as
t-test or linear models), it is often necessary to log-transform the
data. The figure below (left) show that how our data is mainly composed
of small values with a long tail of larger ones, which is a typical
pattern of quantitative omics data.</p>
<p>Below, we use the <code>logTransform</code> function to
log2-transform our data. This time, instead of overwriting the peptides
assay, we are going to create a new one to contain the log2-transformed
data.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-class.html">addAssay</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>         <span class="fu"><a href="../reference/QFeatures-processing.html">logTransform</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>         name <span class="op">=</span> <span class="st">"peptides_log"</span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/QFeatures-class.html">addAssay()</a></code> function is the general function that
adds new assays to a <code>QFeatures</code> object. The step above could
more easily be exectuted with the <code><a href="../reference/QFeatures-processing.html">logTransform()</a></code> method,
that directly returns an updated <code>QFeatures</code> object. Using
<code><a href="../reference/QFeatures-processing.html">logTransform()</a></code> also automatically adds links between
assays.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-processing.html">logTransform</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>                      i <span class="op">=</span> <span class="st">"peptides"</span>,</span>
<span>                      name <span class="op">=</span> <span class="st">"peptides_log"</span><span class="op">)</span></span>
<span><span class="va">cptac</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 2 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [2] peptides_log: SummarizedExperiment with 4051 rows and 6 columns</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html" class="external-link">plotDensities</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html" class="external-link">plotDensities</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Processing_files/figure-html/unnamed-chunk-9-1.png" alt="Quantitative data in its original scale (left) and log2-transformed (right)." width="700"><p class="caption">
Quantitative data in its original scale (left) and log2-transformed
(right).
</p>
</div>
</div>
<div class="section level2">
<h2 id="normalisation">Normalisation<a class="anchor" aria-label="anchor" href="#normalisation"></a>
</h2>
<p>Assays in <code>QFeatures</code> objects can be normalised with the
<code>normalize</code> function. The type of normalisation is defined by
the <code>method</code> argument; below, we use median normalisation,
store the normalised data into a new experiment, and visualise the
resulting data.</p>
<p>The <code><a href="../reference/QFeatures-processing.html">normalize()</a></code> function can also be directly applied to
the <code>QFeatures</code> object.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-processing.html">normalize</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>                   i <span class="op">=</span> <span class="st">"peptides_log"</span>,</span>
<span>                   name <span class="op">=</span> <span class="st">"peptides_norm"</span>,</span>
<span>                   method <span class="op">=</span> <span class="st">"diff.median"</span><span class="op">)</span></span>
<span><span class="va">cptac</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [2] peptides_log: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [3] peptides_norm: SummarizedExperiment with 4051 rows and 6 columns</span></span></code></pre>
<p>It is also possible to extract and normalise the
<code>peptides_log</code> <code>SummarizedExperiment</code> and add it
back to the <code>QFeatures</code> object with
<code><a href="../reference/QFeatures-class.html">addAssay()</a></code>.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-class.html">addAssay</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>         <span class="fu"><a href="../reference/QFeatures-processing.html">normalize</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"peptides_log"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                   method <span class="op">=</span> <span class="st">"center.median"</span><span class="op">)</span>,</span>
<span>         name <span class="op">=</span> <span class="st">"peptides_norm"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html" class="external-link">plotDensities</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"peptides_log"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html" class="external-link">plotDensities</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"peptides_norm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Processing_files/figure-html/unnamed-chunk-10-1.png" alt="Distribution of log2 peptide intensities before (left) and after (right) median normalisation." width="700"><p class="caption">
Distribution of log2 peptide intensities before (left) and after (right)
median normalisation.
</p>
</div>
</div>
<div class="section level2">
<h2 id="feature-aggregation">Feature aggregation<a class="anchor" aria-label="anchor" href="#feature-aggregation"></a>
</h2>
<p>At this stage, it is possible to directly use the peptide-level
intensities to perform a statistical analysis <span class="citation">(Goeminne, Gevaert, and Clement 2016)</span>, or
aggregate the peptide-level data into protein intensities, and perform
the differential expression analysis at the protein level.</p>
<p>To aggregate feature data, we can use the
<code>aggregateFeatures</code> function that takes the following
inputs:</p>
<ul>
<li>the name of the <code>QFeatures</code> instance that contains the
peptide quantitation data - <code>"cptac"</code> in our example;</li>
<li>
<strong><code>i</code></strong>: the name or index of the assay that
contains the (normalised) peptide quantitation data -
<code>"peptides_norm"</code> in our case;</li>
<li>
<strong><code>fcol</code></strong>: the feature variable (in the
assay above) to be used to define what peptides to aggregate -
<code>"Proteins"</code> here, given that we want to aggregate all
peptides that belong to one protein (group);</li>
<li>
<strong><code>name</code></strong>: the name of the new aggregates
assay - <code>"proteins"</code> in this case;</li>
<li>and finally <strong><code>fun</code></strong>, the function that
will compute this aggregation - we will be using the default value,
namely <code>robustSummary</code> <span class="citation">(Sticker et al.
2019)</span>.</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">cptac</span>,</span>
<span>                           i <span class="op">=</span> <span class="st">"peptides_norm"</span>,</span>
<span>                           fcol <span class="op">=</span> <span class="st">"Proteins"</span>,</span>
<span>                           name <span class="op">=</span> <span class="st">"proteins"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Aggregated: 1/1</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cptac</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 4 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] peptides: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [2] peptides_log: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [3] peptides_norm: SummarizedExperiment with 4051 rows and 6 columns </span></span>
<span><span class="co">##  [4] proteins: SummarizedExperiment with 1125 rows and 6 columns</span></span></code></pre>
<p>We obtain a final 1125 quantified proteins in the new
<code>proteins</code> assay. Below, we display the quantitation data for
the first 6 proteins and their respective variables. The latter shown
that number of peptides that were using during the aggregation step
(<code>.n</code> column).</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      Intensity.6A_7 Intensity.6A_8</span></span>
<span><span class="co">## P00918ups|CAH2_HUMAN_UPS                   17.23988       16.98222</span></span>
<span><span class="co">## P01008ups|ANT3_HUMAN_UPS;CON__P41361       16.81917       16.11327</span></span>
<span><span class="co">## P01127ups|PDGFB_HUMAN_UPS                  16.45163       16.90199</span></span>
<span><span class="co">## P02144ups|MYG_HUMAN_UPS                    16.81662       16.55897</span></span>
<span><span class="co">## P02753ups|RETBP_HUMAN_UPS                  17.80433       16.79555</span></span>
<span><span class="co">## P02787ups|TRFE_HUMAN_UPS                   16.74488       16.97394</span></span>
<span><span class="co">##                                      Intensity.6A_9 Intensity.6B_7</span></span>
<span><span class="co">## P00918ups|CAH2_HUMAN_UPS                   16.63167       18.27738</span></span>
<span><span class="co">## P01008ups|ANT3_HUMAN_UPS;CON__P41361       16.33382       16.72030</span></span>
<span><span class="co">## P01127ups|PDGFB_HUMAN_UPS                  16.83464       18.19830</span></span>
<span><span class="co">## P02144ups|MYG_HUMAN_UPS                    17.28038       17.86570</span></span>
<span><span class="co">## P02753ups|RETBP_HUMAN_UPS                  16.55527       18.39382</span></span>
<span><span class="co">## P02787ups|TRFE_HUMAN_UPS                   16.34641       18.13812</span></span>
<span><span class="co">##                                      Intensity.6B_8 Intensity.6B_9</span></span>
<span><span class="co">## P00918ups|CAH2_HUMAN_UPS                   18.54886       18.46754</span></span>
<span><span class="co">## P01008ups|ANT3_HUMAN_UPS;CON__P41361       16.74130       16.48097</span></span>
<span><span class="co">## P01127ups|PDGFB_HUMAN_UPS                  18.77132       17.16724</span></span>
<span><span class="co">## P02144ups|MYG_HUMAN_UPS                    18.55434       18.29205</span></span>
<span><span class="co">## P02753ups|RETBP_HUMAN_UPS                  17.73507       18.15238</span></span>
<span><span class="co">## P02787ups|TRFE_HUMAN_UPS                   18.51059       18.15718</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1125 rows and 3 columns</span></span>
<span><span class="co">##                                           Proteins Leading.razor.protein</span></span>
<span><span class="co">##                                        &lt;character&gt;           &lt;character&gt;</span></span>
<span><span class="co">## P00918ups|CAH2_HUMAN_UPS             P00918ups|...         P00918ups|...</span></span>
<span><span class="co">## P01008ups|ANT3_HUMAN_UPS;CON__P41361 P01008ups|...         P01008ups|...</span></span>
<span><span class="co">## P01127ups|PDGFB_HUMAN_UPS            P01127ups|...         P01127ups|...</span></span>
<span><span class="co">## P02144ups|MYG_HUMAN_UPS              P02144ups|...         P02144ups|...</span></span>
<span><span class="co">## P02753ups|RETBP_HUMAN_UPS            P02753ups|...         P02753ups|...</span></span>
<span><span class="co">## ...                                            ...                   ...</span></span>
<span><span class="co">## sp|Q99207|NOP14_YEAST                sp|Q99207|...         sp|Q99207|...</span></span>
<span><span class="co">## sp|Q99216|PNO1_YEAST                 sp|Q99216|...         sp|Q99216|...</span></span>
<span><span class="co">## sp|Q99257|MEX67_YEAST                sp|Q99257|...         sp|Q99257|...</span></span>
<span><span class="co">## sp|Q99258|RIB3_YEAST                 sp|Q99258|...         sp|Q99258|...</span></span>
<span><span class="co">## sp|Q99383|HRP1_YEAST                 sp|Q99383|...         sp|Q99383|...</span></span>
<span><span class="co">##                                             .n</span></span>
<span><span class="co">##                                      &lt;integer&gt;</span></span>
<span><span class="co">## P00918ups|CAH2_HUMAN_UPS                     1</span></span>
<span><span class="co">## P01008ups|ANT3_HUMAN_UPS;CON__P41361         1</span></span>
<span><span class="co">## P01127ups|PDGFB_HUMAN_UPS                    1</span></span>
<span><span class="co">## P02144ups|MYG_HUMAN_UPS                      1</span></span>
<span><span class="co">## P02753ups|RETBP_HUMAN_UPS                    2</span></span>
<span><span class="co">## ...                                        ...</span></span>
<span><span class="co">## sp|Q99207|NOP14_YEAST                        1</span></span>
<span><span class="co">## sp|Q99216|PNO1_YEAST                         1</span></span>
<span><span class="co">## sp|Q99257|MEX67_YEAST                        2</span></span>
<span><span class="co">## sp|Q99258|RIB3_YEAST                         2</span></span>
<span><span class="co">## sp|Q99383|HRP1_YEAST                         2</span></span></code></pre>
<p>We can get a quick overview of this <code>.n</code> variable by
computing the table below, that shows us that we have 405 proteins that
are based on a single peptides, 230 that are based on two, 119 that are
based on three, … and a single protein that is the results of
aggregating 44 peptides.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">cptac</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">.n</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 </span></span>
<span><span class="co">## 405 230 119  84  64  53  37  29  24  24  13   9   4   3   3   7   3   1   1   1 </span></span>
<span><span class="co">##  21  22  23  24  25  30  31  33  44 </span></span>
<span><span class="co">##   1   2   2   1   1   1   1   1   1</span></span></code></pre>
<p>Let’s choose <code>P02787ups|TRFE_HUMAN_UPS</code> and visualise its
expression pattern in the 2 groups at the protein and (log-tranformed
and normalised) peptide level. We drop the first peptide-level assay as
it is on a different scale (i.e. not log-transformed).</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/QFeatures-longForm.html">longForm</a></span><span class="op">(</span><span class="va">cptac</span><span class="op">[</span><span class="st">"P02787ups|TRFE_HUMAN_UPS"</span>, , <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="va">colname</span><span class="op">)</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"Intensity\\."</span>, <span class="st">""</span>, <span class="va">colname</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="va">value</span>,</span>
<span>               colour <span class="op">=</span> <span class="va">rowname</span>,</span>
<span>               shape <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">rowname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">assay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'drops()'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 6 sampleMap rows not in names(experiments)</span></span></code></pre>
<div class="figure">
<img src="Processing_files/figure-html/unnamed-chunk-13-1.png" alt="Expression intensities for the protein *P02787ups|TRFE_HUMAN_UPS* (right, green) and its peptides (left) in groups A (circles) and B (triangles)." width="700"><p class="caption">
Expression intensities for the protein <em>P02787ups|TRFE_HUMAN_UPS</em>
(right, green) and its peptides (left) in groups A (circles) and B
(triangles).
</p>
</div>
</div>
<div class="section level2">
<h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<ul>
<li>The other vignettes in the <code>QFeatures</code> package.</li>
<li>The <a href="https://lgatto.github.io/QFeaturesWorkshop2020/index.html" class="external-link">QFeaturesWorkshop2020</a>
workshop, presented at the EuroBioc2020 meeting. It also documents how
to use a custom docker container to run the workshop code.</li>
<li>The <a href="https://uclouvain-cbio.github.io/WSBIM2122/sec-prot.html" class="external-link">Quantitative
proteomics data analysis</a> chapter of the WSBIM2122 course.</li>
</ul>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] gplots_3.2.0                dplyr_1.1.4                </span></span>
<span><span class="co">##  [3] ggplot2_3.5.2               QFeatures_1.19.2           </span></span>
<span><span class="co">##  [5] MultiAssayExperiment_1.35.3 SummarizedExperiment_1.39.0</span></span>
<span><span class="co">##  [7] Biobase_2.69.0              GenomicRanges_1.61.0       </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.45.3         IRanges_2.43.0             </span></span>
<span><span class="co">## [11] S4Vectors_0.47.0            BiocGenerics_0.55.0        </span></span>
<span><span class="co">## [13] generics_0.1.4              MatrixGenerics_1.21.0      </span></span>
<span><span class="co">## [15] matrixStats_1.5.0           BiocStyle_2.37.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] gtable_0.3.6            xfun_0.52               bslib_0.9.0            </span></span>
<span><span class="co">##  [4] caTools_1.18.3          htmlwidgets_1.6.4       lattice_0.22-7         </span></span>
<span><span class="co">##  [7] bitops_1.0-9            vctrs_0.6.5             tools_4.5.0            </span></span>
<span><span class="co">## [10] tibble_3.2.1            cluster_2.1.8.1         pkgconfig_2.0.3        </span></span>
<span><span class="co">## [13] BiocBaseUtils_1.11.0    KernSmooth_2.23-26      Matrix_1.7-3           </span></span>
<span><span class="co">## [16] RColorBrewer_1.1-3      desc_1.4.3              lifecycle_1.0.4        </span></span>
<span><span class="co">## [19] farver_2.1.2            compiler_4.5.0          stringr_1.5.1          </span></span>
<span><span class="co">## [22] textshaping_1.0.1       statmod_1.5.0           clue_0.3-66            </span></span>
<span><span class="co">## [25] htmltools_0.5.8.1       sass_0.4.10             yaml_2.3.10            </span></span>
<span><span class="co">## [28] lazyeval_0.2.2          pkgdown_2.1.3.9000      pillar_1.10.2          </span></span>
<span><span class="co">## [31] crayon_1.5.3            jquerylib_0.1.4         tidyr_1.3.1            </span></span>
<span><span class="co">## [34] MASS_7.3-65             limma_3.65.1            DelayedArray_0.35.1    </span></span>
<span><span class="co">## [37] cachem_1.1.0            abind_1.4-8             gtools_3.9.5           </span></span>
<span><span class="co">## [40] tidyselect_1.2.1        digest_0.6.37           stringi_1.8.7          </span></span>
<span><span class="co">## [43] purrr_1.0.4             reshape2_1.4.4          bookdown_0.43          </span></span>
<span><span class="co">## [46] labeling_0.4.3          fastmap_1.2.0           grid_4.5.0             </span></span>
<span><span class="co">## [49] cli_3.6.5               SparseArray_1.9.0       magrittr_2.0.3         </span></span>
<span><span class="co">## [52] S4Arrays_1.9.0          withr_3.0.2             scales_1.4.0           </span></span>
<span><span class="co">## [55] UCSC.utils_1.5.0        rmarkdown_2.29          XVector_0.49.0         </span></span>
<span><span class="co">## [58] httr_1.4.7              igraph_2.1.4            ragg_1.4.0             </span></span>
<span><span class="co">## [61] evaluate_1.0.3          knitr_1.50              rlang_1.1.6            </span></span>
<span><span class="co">## [64] Rcpp_1.0.14             glue_1.8.0              BiocManager_1.30.25    </span></span>
<span><span class="co">## [67] msdata_0.49.0           jsonlite_2.0.0          AnnotationFilter_1.33.0</span></span>
<span><span class="co">## [70] R6_2.6.1                plyr_1.8.9              systemfonts_1.2.3      </span></span>
<span><span class="co">## [73] fs_1.6.6                ProtGenerics_1.41.0     MsCoreUtils_1.21.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>This vignette is distributed under a <a href="https://creativecommons.org/licenses/by-sa/2.0/" class="external-link">CC BY-SA
license</a> license.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Cox:2008" class="csl-entry">
Cox, J, and M Mann. 2008. <span>“MaxQuant Enables High Peptide
Identification Rates, Individualized p.p.b.-Range Mass Accuracies and
Proteome-Wide Protein Quantification.”</span> <em>Nat Biotechnol</em> 26
(12): 1367–72. <a href="https://doi.org/10.1038/nbt.1511" class="external-link">https://doi.org/10.1038/nbt.1511</a>.
</div>
<div id="ref-Goeminne:2016" class="csl-entry">
Goeminne, L J, K Gevaert, and L Clement. 2016. <span>“Peptide-Level
Robust Ridge Regression Improves Estimation, Sensitivity, and
Specificity in Data-Dependent Quantitative Label-Free Shotgun
Proteomics.”</span> <em>Mol Cell Proteomics</em> 15 (2): 657–68. <a href="https://doi.org/10.1074/mcp.M115.055897" class="external-link">https://doi.org/10.1074/mcp.M115.055897</a>.
</div>
<div id="ref-Lazar:2016" class="csl-entry">
Lazar, C, L Gatto, M Ferro, C Bruley, and T Burger. 2016.
<span>“Accounting for the Multiple Natures of Missing Values in
Label-Free Quantitative Proteomics Data Sets to Compare Imputation
Strategies.”</span> <em>J Proteome Res</em> 15 (4): 1116–25. <a href="https://doi.org/10.1021/acs.jproteome.5b00981" class="external-link">https://doi.org/10.1021/acs.jproteome.5b00981</a>.
</div>
<div id="ref-Paulovich:2010" class="csl-entry">
Paulovich, Amanda G, Dean Billheimer, Amy-Joan L Ham, Lorenzo
Vega-Montoto, Paul A Rudnick, David L Tabb, Pei Wang, et al. 2010.
<span>“Interlaboratory Study Characterizing a Yeast Performance Standard
for Benchmarking <span>LC-MS</span> Platform Performance.”</span>
<em>Mol. Cell. Proteomics</em> 9 (2): 242–54.
</div>
<div id="ref-Sticker:2019" class="csl-entry">
Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement.
2019. <span>“Robust Summarization and Inference in Proteome-Wide
Label-Free Quantification.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/668863" class="external-link">https://doi.org/10.1101/668863</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Laurent Gatto, Christophe Vanderaa.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
