<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Quantitative features for mass spectrometry data • QFeatures</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Quantitative features for mass spectrometry data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QFeatures</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.19.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/QFeatures.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Processing.html">Processing quantitative proteomics data with QFeatures</a></li>
    <li><a class="dropdown-item" href="../articles/read_QFeatures.html">Load mass spectrometry-based proteomics data using `readQFeatures()`</a></li>
    <li><a class="dropdown-item" href="../articles/Visualization.html">Data visualization from a QFeatures object</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/rformassspectrometry/QFeatures/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Quantitative features for mass spectrometry data</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/rformassspectrometry/QFeatures/blob/master/vignettes/QFeatures.Rmd" class="external-link"><code>vignettes/QFeatures.Rmd</code></a></small>
      <div class="d-none name"><code>QFeatures.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes the functionality implemented in the
      QFeatures package. QFeatures provides infrastructure to manage and
      process quantitative features for high-throughput mass
      spectrometry assays, including proteomics and metabolomics. This
      vignette is distributed under a CC BY-SA license.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>QFeatures</code> package provides infrastructure (that is
classes to store data and the methods to process and manipulate them) to
manage and analyse quantitative features from mass spectrometry
experiments. It is based on the <code>SummarizedExperiment</code> and
<code>MultiAssayExperiment</code> classes. Assays in a QFeatures object
have a hierarchical relation: proteins are composed of peptides,
themselves produced by spectra, as depicted in figure
@ref(fig:featuresplot). Throughout the aggregation and processing of
these data, the relations between assays are tracked and recorded, thus
allowing users to easily navigate across spectra, peptide and protein
quantitative data.</p>
<div class="figure">
<img src="QFeatures_files/figure-html/featuresplot-1.png" class="r-plt" alt="Conceptual representation of a `QFeatures` object and the aggregative relation between different assays." width="700"><p class="caption">
Conceptual representation of a <code>QFeatures</code> object and the
aggregative relation between different assays.
</p>
</div>
<p>In the following sections, we are going to demonstrate how to create
a single-assay <code>QFeatures</code> objects starting from a
spreadsheet, how to compute the next assays (peptides and proteins), and
how these can be manipulated and explored.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://rformassspectrometry.github.io/QFeatures">"QFeatures"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-qfeatures-object">Creating <code>QFeatures</code> object<a class="anchor" aria-label="anchor" href="#creating-qfeatures-object"></a>
</h2>
<p>While <code>QFeatures</code> objects can be created manually (see
<code><a href="../reference/QFeatures-class.html">?QFeatures</a></code> for details), most users will probably possess
quantitative data in a spreadsheet or a dataframe. In such cases, the
easiest is to use the <code>readQFeatures</code> function to extract the
quantitative data and metadata columns. Below, we load the
<code>hlpsms</code> dataframe that contains data for 3010 PSMs from the
TMT-10plex hyperLOPIT spatial proteomics experiment from <span class="citation">(Christoforou et al. 2016)</span>. The
<code>quantCols</code> argument specifies that columns 1 to 10 contain
quantitation data, and that the assay should be named <code>psms</code>
in the returned <code>QFeatures</code> object, to reflect the nature of
the data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hlpsms</span><span class="op">)</span></span>
<span><span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readQFeatures.html">readQFeatures</a></span><span class="op">(</span><span class="va">hlpsms</span>, quantCols <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, name <span class="op">=</span> <span class="st">"psms"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Checking arguments.</span></span></code></pre>
<pre><code><span><span class="co">## Loading data as a 'SummarizedExperiment' object.</span></span></code></pre>
<pre><code><span><span class="co">## Formatting sample annotations (colData).</span></span></code></pre>
<pre><code><span><span class="co">## Formatting data as a 'QFeatures' object.</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 1 set:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 3010 rows and 10 columns</span></span></code></pre>
<p>Below, we see that we can extract an assay using its index or its
name. The individual assays are stored as <em>SummarizedExperiment</em>
object and further access its quantitative data and metadata using the
<code>assay</code> and <code>rowData</code> functions</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 3010 10 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(3010): 1 2 ... 3009 3010</span></span>
<span><span class="co">## rowData names(18): Sequence ProteinDescriptions ... RTmin markers</span></span>
<span><span class="co">## colnames(10): X126 X127C ... X130N X131</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span><span class="op">[[</span><span class="st">"psms"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 3010 10 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(3010): 1 2 ... 3009 3010</span></span>
<span><span class="co">## rowData names(18): Sequence ProteinDescriptions ... RTmin markers</span></span>
<span><span class="co">## colnames(10): X126 X127C ... X130N X131</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">hl</span><span class="op">[[</span><span class="st">"psms"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         X126      X127C       X127N      X128C       X128N      X129C</span></span>
<span><span class="co">## 1 0.12283431 0.08045915 0.070804055 0.09386901 0.051815695 0.13034383</span></span>
<span><span class="co">## 2 0.35268185 0.14162381 0.167523880 0.07843497 0.071087436 0.03214548</span></span>
<span><span class="co">## 3 0.01546089 0.16142297 0.086938133 0.23120844 0.114664348 0.09610188</span></span>
<span><span class="co">## 4 0.04702854 0.09288723 0.102012167 0.11125409 0.067969116 0.14155358</span></span>
<span><span class="co">## 5 0.01044693 0.15866147 0.167315736 0.21017494 0.147946673 0.07088253</span></span>
<span><span class="co">## 6 0.04955362 0.01215244 0.002477681 0.01297833 0.002988949 0.06253195</span></span>
<span><span class="co">##        X129N       X130C      X130N       X131</span></span>
<span><span class="co">## 1 0.17540095 0.040068658 0.11478839 0.11961594</span></span>
<span><span class="co">## 2 0.06686260 0.031961793 0.02810434 0.02957384</span></span>
<span><span class="co">## 3 0.15977819 0.010127118 0.08059400 0.04370403</span></span>
<span><span class="co">## 4 0.18015910 0.035329902 0.12166589 0.10014038</span></span>
<span><span class="co">## 5 0.17555789 0.007088253 0.02884754 0.02307803</span></span>
<span><span class="co">## 6 0.01726511 0.172651119 0.37007905 0.29732174</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">[[</span><span class="st">"psms"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 18 columns</span></span>
<span><span class="co">##      Sequence ProteinDescriptions NbProteins ProteinGroupAccessions</span></span>
<span><span class="co">##   &lt;character&gt;         &lt;character&gt;  &lt;integer&gt;            &lt;character&gt;</span></span>
<span><span class="co">## 1     SQGEIDk       Tetratrico...          1                 Q8BYY4</span></span>
<span><span class="co">## 2     YEAQGDk       Vacuolar p...          1                 P46467</span></span>
<span><span class="co">## 3     TTScDTk       C-type man...          1                 Q64449</span></span>
<span><span class="co">## 4     aEELESR       Liprin-alp...          1                 P60469</span></span>
<span><span class="co">## 5     aQEEAIk       Isoform 2 ...          2               P13597-2</span></span>
<span><span class="co">## 6    dGAVDGcR       Structural...          1                 Q6P5D8</span></span>
<span><span class="co">##   Modifications    qValue       PEP  IonScore NbMissedCleavages</span></span>
<span><span class="co">##     &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;         &lt;integer&gt;</span></span>
<span><span class="co">## 1 K7(TMT6ple...     0.008   0.11800        27                 0</span></span>
<span><span class="co">## 2 K7(TMT6ple...     0.001   0.01070        27                 0</span></span>
<span><span class="co">## 3 C4(Carbami...     0.008   0.11800        11                 0</span></span>
<span><span class="co">## 4 N-Term(TMT...     0.002   0.04450        24                 0</span></span>
<span><span class="co">## 5 N-Term(Car...     0.001   0.00850        36                 0</span></span>
<span><span class="co">## 6 N-Term(TMT...     0.000   0.00322        26                 0</span></span>
<span><span class="co">##   IsolationInterference IonInjectTimems Intensity    Charge      mzDa      MHDa</span></span>
<span><span class="co">##               &lt;integer&gt;       &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1                     0              70    335000         2   503.274   1005.54</span></span>
<span><span class="co">## 2                     0              70    926000         2   520.267   1039.53</span></span>
<span><span class="co">## 3                     0              70    159000         2   521.258   1041.51</span></span>
<span><span class="co">## 4                     0              70    232000         2   531.785   1062.56</span></span>
<span><span class="co">## 5                     0              70    212000         2   537.804   1074.60</span></span>
<span><span class="co">## 6                     0              70    865000         2   539.761   1078.51</span></span>
<span><span class="co">##   DeltaMassPPM     RTmin       markers</span></span>
<span><span class="co">##      &lt;numeric&gt; &lt;numeric&gt;   &lt;character&gt;</span></span>
<span><span class="co">## 1        -0.38     24.02       unknown</span></span>
<span><span class="co">## 2         0.61     18.85       unknown</span></span>
<span><span class="co">## 3         1.11     10.17       unknown</span></span>
<span><span class="co">## 4         0.35     29.18       unknown</span></span>
<span><span class="co">## 5         1.70     25.56 Plasma mem...</span></span>
<span><span class="co">## 6        -0.67     21.27 Nucleus - ...</span></span></code></pre>
<p>For further details on how to manipulate such objects, refer to the
<em><a href="https://bioconductor.org/packages/3.22/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
<span class="citation">(Ramos et al. 2017)</span> and <em><a href="https://bioconductor.org/packages/3.22/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
<span class="citation">(Morgan et al. 2019)</span> packages.</p>
<p>As illustrated in figure @ref(fig:featuresplot), an central
characteristic of <code>QFeatures</code> objects is the aggregative
relation between their assays. This can be obtained with the
<code>aggregateFeatures</code> function that will aggregate quantitative
features from one assay into a new one. In the next code chunk, we
aggregate PSM-level data into peptide by grouping all PSMs that were
matched the same peptide sequence. Below, the aggregation function is
set, as an example, to the mean. The new assay is named
<em>peptides</em>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="st">"psms"</span>, <span class="st">"Sequence"</span>,</span>
<span>                        name <span class="op">=</span> <span class="st">"peptides"</span>, fun <span class="op">=</span> <span class="va">colMeans</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Your row data contain missing values. Please read the relevant</span></span>
<span><span class="co">## section(s) in the aggregateFeatures manual page regarding the effects</span></span>
<span><span class="co">## of missing values on data aggregation.</span></span></code></pre>
<pre><code><span><span class="co">## Aggregated: 1/1</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 2 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 3010 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 2923 rows and 10 columns</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span><span class="op">[[</span><span class="st">"peptides"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2923 10 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(2): assay aggcounts</span></span>
<span><span class="co">## rownames(2923): aAADADVSLHNFSAR aAADPSEk ... yYSFFDLNPk YYSIASSSk</span></span>
<span><span class="co">## rowData names(7): Sequence ProteinDescriptions ... markers .n</span></span>
<span><span class="co">## colnames(10): X126 X127C ... X130N X131</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<p>Below, we repeat the aggregation operation by grouping peptides into
proteins as defined by the <em>ProteinGroupAccessions</em> variable.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="st">"peptides"</span>, <span class="st">"ProteinGroupAccessions"</span>,</span>
<span>                        name <span class="op">=</span> <span class="st">"proteins"</span>, fun <span class="op">=</span> <span class="va">colMeans</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Aggregated: 1/1</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 3010 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 2923 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 1596 rows and 10 columns</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 1596 10 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(2): assay aggcounts</span></span>
<span><span class="co">## rownames(1596): A2A432 A2A6Q5-3 ... Q9Z2Z9 Q9Z315</span></span>
<span><span class="co">## rowData names(3): ProteinGroupAccessions markers .n</span></span>
<span><span class="co">## colnames(10): X126 X127C ... X130N X131</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<p>The sample assayed in a <code>QFeatures</code> object can be
documented in the <code>colData</code> slot. The <code>hl</code> data
doens’t currently possess any sample metadata. These can be addedd as a
new <code>DataFrame</code> with matching names (i.e. the
<code>DataFrame</code> rownames must be identical assay’s colnames) or
can be added one variable at at time, as shown below.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 10 rows and 0 columns</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span><span class="op">$</span><span class="va">tag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"126"</span>, <span class="st">"127N"</span>, <span class="st">"127C"</span>, <span class="st">"128N"</span>, <span class="st">"128C"</span>, <span class="st">"129N"</span>, <span class="st">"129C"</span>,</span>
<span>            <span class="st">"130N"</span>, <span class="st">"130C"</span>, <span class="st">"131"</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 10 rows and 1 column</span></span>
<span><span class="co">##               tag</span></span>
<span><span class="co">##       &lt;character&gt;</span></span>
<span><span class="co">## X126          126</span></span>
<span><span class="co">## X127C        127N</span></span>
<span><span class="co">## X127N        127C</span></span>
<span><span class="co">## X128C        128N</span></span>
<span><span class="co">## X128N        128C</span></span>
<span><span class="co">## X129C        129N</span></span>
<span><span class="co">## X129N        129C</span></span>
<span><span class="co">## X130C        130N</span></span>
<span><span class="co">## X130N        130C</span></span>
<span><span class="co">## X131          131</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="manipulating-feature-metadata">Manipulating feature metadata<a class="anchor" aria-label="anchor" href="#manipulating-feature-metadata"></a>
</h2>
<p>The <code>QFeatures</code> package provides some utility functions
that streamline the accession and manipulation of the feature
metadata.</p>
<p>The feature metadata, more generally referred to as
<code>rowData</code> in the Bioconductor ecosystem, is specific to each
assay in a <code>QFeatures</code> object. Therefore there are as many
<code>rowData</code> tables as there are assays.
<code>rowDataNames</code> provides a list where each element contains
the name of the <code>rowData</code> columns available in the
corresponding assay.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-class.html">rowDataNames</a></span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 3</span></span>
<span><span class="co">## [["psms"]] Sequence ProteinDescriptions NbProteins ... RTmin markers</span></span>
<span><span class="co">## [["peptides"]] Sequence ProteinDescriptions NbProteins ... markers .n</span></span>
<span><span class="co">## [["proteins"]] ProteinGroupAccessions markers .n</span></span></code></pre>
<p>We saw above how to get the <code>rowData</code> from an assay, but
we can also extract the <code>rowData</code> for all assays by calling
the function on the <code>QFeautures</code> object directly. Similarly
to <code>rowDataNames</code>, a list is returned where each element
contains the <code>rowData</code> available in the corresponding
assay.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrameList of length 3</span></span>
<span><span class="co">## names(3): psms peptides proteins</span></span></code></pre>
<p>In some cases, we are interested in extracting the
<code>rowData</code> as a single data table. This is easily performed
using the <code>rbindRowData</code> function. The function will
automatically select the columns that are common to all selected
assays.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-class.html">rbindRowData</a></span><span class="op">(</span><span class="va">hl</span>, i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"peptides"</span>, <span class="st">"proteins"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 4519 rows and 5 columns</span></span>
<span><span class="co">##            assay       rowname ProteinGroupAccessions       markers        .n</span></span>
<span><span class="co">##      &lt;character&gt;   &lt;character&gt;            &lt;character&gt;   &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## 1       peptides aAADADVSLH...                 P0C7N9       unknown         1</span></span>
<span><span class="co">## 2       peptides      aAADPSEk                 Q6P9Q6       unknown         1</span></span>
<span><span class="co">## 3       peptides aAAQFLTSTW...                 Q9D7Z3 Nucleus - ...         1</span></span>
<span><span class="co">## 4       peptides aAASGAAGDk...                 A2AGT5       unknown         1</span></span>
<span><span class="co">## 5       peptides     AAAVSTEGk                 Q9ERE8       unknown         1</span></span>
<span><span class="co">## ...          ...           ...                    ...           ...       ...</span></span>
<span><span class="co">## 4515    proteins        Q9Z2V5                 Q9Z2V5       unknown         1</span></span>
<span><span class="co">## 4516    proteins        Q9Z2W0                 Q9Z2W0       unknown         1</span></span>
<span><span class="co">## 4517    proteins        Q9Z2X1                 Q9Z2X1       unknown         1</span></span>
<span><span class="co">## 4518    proteins        Q9Z2Z9                 Q9Z2Z9       unknown         3</span></span>
<span><span class="co">## 4519    proteins        Q9Z315                 Q9Z315       unknown         1</span></span></code></pre>
<p>We can also replace and add columns in the <code>rowData</code>. This
requires to provide a <code>List</code> where the names of the
<code>List</code> point to the assay to be updated and the elements of
the <code>List</code> contain <code>DataFrame</code>s with the
replacement values. If the <code>DataFrame</code> contains a column that
is not present in the <code>rowData</code>, that column will get added
to the <code>rowData</code>. For instance, let’s add a
<code>rowData</code> variables with the mean protein expression as well
as the associated standard deviation. First, we need to create the
<code>DataFrame</code> with the mean expression.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span>mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">hl</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                sd <span class="op">=</span> <span class="fu">rowSds</span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">hl</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then, we create the list and name the element <code>proteins</code>
so that the new data is added to the <code>rowData</code> of the
<code>proteins</code> assay. To add the list, we insert it back into the
<code>rowData</code>.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/List-class.html" class="external-link">List</a></span><span class="op">(</span>proteins <span class="op">=</span> <span class="va">dF</span><span class="op">)</span></span></code></pre></div>
<p>As shown below, the new <code>mean</code> and <code>sd</code>
variables have been added to the <code>rowData</code> of the
<code>proteins</code> assay.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1596 rows and 5 columns</span></span>
<span><span class="co">##          ProteinGroupAccessions     markers        .n      mean        sd</span></span>
<span><span class="co">##                     &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## A2A432                   A2A432     unknown         1         1 0.0822395</span></span>
<span><span class="co">## A2A6Q5-3               A2A6Q5-3     unknown         1         1 0.0891478</span></span>
<span><span class="co">## A2A8L5                   A2A8L5     unknown         2         1 0.1009041</span></span>
<span><span class="co">## A2AF47                   A2AF47     unknown         1         1 0.0749159</span></span>
<span><span class="co">## A2AGT5                   A2AGT5     unknown         6         1 0.1065126</span></span>
<span><span class="co">## ...                         ...         ...       ...       ...       ...</span></span>
<span><span class="co">## Q9Z2V5                   Q9Z2V5     unknown         1         1 0.0882136</span></span>
<span><span class="co">## Q9Z2W0                   Q9Z2W0     unknown         1         1 0.0565321</span></span>
<span><span class="co">## Q9Z2X1                   Q9Z2X1     unknown         1         1 0.1539930</span></span>
<span><span class="co">## Q9Z2Z9                   Q9Z2Z9     unknown         3         1 0.0930030</span></span>
<span><span class="co">## Q9Z315                   Q9Z315     unknown         1         1 0.1234534</span></span></code></pre>
<p>Note that you can also replace an existing column in the
<code>rowData</code> by naming the column name in the
<code>DataFrame</code> after the column to replace.</p>
</div>
<div class="section level2">
<h2 id="subsetting">Subsetting<a class="anchor" aria-label="anchor" href="#subsetting"></a>
</h2>
<p>One particularity of the <code>QFeatures</code> infrastructure is
that the features of the constitutive assays are linked through an
aggregative relation. This relation is recorded when creating new assays
with <code>aggregateFeatures</code> and is exploited when subsetting
<code>QFeature</code> by their feature names.</p>
<p>In the example below, we are interested in the Stat3B isoform of the
Signal transducer and activator of transcription 3 (STAT3) with
accession number P42227-2. This accession number corresponds to a
feature name in the <em>proteins</em> assay. But this protein row was
computed from 8 peptide rows in the <em>peptides</em> assay, themselves
resulting from the aggregation of 8 rows in the <em>psms</em> assay.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat3</span> <span class="op">&lt;-</span> <span class="va">hl</span><span class="op">[</span><span class="st">"P42227-2"</span>, , <span class="op">]</span></span>
<span><span class="va">stat3</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 9 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 8 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 1 rows and 10 columns</span></span></code></pre>
<p>We can easily visualise this new <em>QFeatures</em> object using
<code>ggplot2</code> once converted into a <code>data.frame</code>. See
the visualization vignette for more details about data exploration from
a <code>QFeatures</code> object.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat3_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="../reference/QFeatures-longForm.html">longForm</a></span><span class="op">(</span><span class="va">stat3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stat3_df</span><span class="op">$</span><span class="va">assay</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">stat3_df</span><span class="op">$</span><span class="va">assay</span>,</span>
<span>                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"psms"</span>, <span class="st">"peptides"</span>, <span class="st">"proteins"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">stat3_df</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">colname</span>,</span>
<span>           y <span class="op">=</span> <span class="va">value</span>,</span>
<span>           group <span class="op">=</span> <span class="va">rowname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">assay</span><span class="op">)</span></span></code></pre></div>
<p><img src="QFeatures_files/figure-html/plotstat3-1.png" class="r-plt" width="700"></p>
<p>Below we repeat the same operation for the Signal transducer and
activator of transcription 1 (STAT1) and 3 (STAT3) accession numbers,
namely P42227-2 and P42225. We obtain a new <code>QFeatures</code>
instance containing 2 proteins, 9 peptides and 10 PSMS. From this, we
can readily conclude that STAT1 was identified by a single
PSM/peptide.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat</span> <span class="op">&lt;-</span> <span class="va">hl</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P42227-2"</span>, <span class="st">"P42225"</span><span class="op">)</span>, , <span class="op">]</span></span>
<span><span class="va">stat</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 10 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 9 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 2 rows and 10 columns</span></span></code></pre>
<p>Below, we visualise the expression profiles for the two proteins.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="../reference/QFeatures-longForm.html">longForm</a></span><span class="op">(</span><span class="va">stat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stat_df</span><span class="op">$</span><span class="va">stat3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">stat_df</span><span class="op">$</span><span class="va">rowname</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">stat3_df</span><span class="op">$</span><span class="va">rowname</span>,</span>
<span>                        <span class="st">"STAT3"</span>, <span class="st">"STAT1"</span><span class="op">)</span></span>
<span><span class="va">stat_df</span><span class="op">$</span><span class="va">assay</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">stat_df</span><span class="op">$</span><span class="va">assay</span>,</span>
<span>                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"psms"</span>, <span class="st">"peptides"</span>, <span class="st">"proteins"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">stat_df</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">colname</span>,</span>
<span>           y <span class="op">=</span> <span class="va">value</span>,</span>
<span>           group <span class="op">=</span> <span class="va">rowname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">stat3</span> <span class="op">~</span> <span class="va">assay</span><span class="op">)</span></span></code></pre></div>
<p><img src="QFeatures_files/figure-html/plotstat-1.png" class="r-plt" width="700"></p>
<p>The subsetting by feature names is also available as a call to the
<code>subsetByFeature</code> function, for use with the pipe
operator.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-subsetBy.html">subsetByFeature</a></span><span class="op">(</span><span class="st">"P42227-2"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 9 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 8 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 1 rows and 10 columns</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-subsetBy.html">subsetByFeature</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P42227-2"</span>, <span class="st">"P42225"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 10 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 9 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 2 rows and 10 columns</span></span></code></pre>
<p>and possibly</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-subsetBy.html">subsetByFeature</a></span><span class="op">(</span><span class="st">"P42227-2"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/QFeatures-longForm.html">longForm</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">colname</span>,</span>
<span>               y <span class="op">=</span> <span class="va">value</span>,</span>
<span>               group <span class="op">=</span> <span class="va">rowname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">assay</span><span class="op">)</span></span></code></pre></div>
<p>to reproduce the line plot.</p>
</div>
<div class="section level2">
<h2 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h2>
<p>QFeatures is assays can also be filtered based on variables in their
respective row data slots using the <code>filterFeatures</code>
function. The filters can be defined using the formula interface or
using <code>AnnotationFilter</code> objects from the
<code>r BiocStyle::Biocpkg("AnnotationFilter")</code> package <span class="citation">(Morgan and Rainer 2019)</span>. In addition to the
pre-defined filters (such as <code>SymbolFilter</code>,
<code>ProteinIdFilter</code>, … that filter on gene symbol, protein
identifier, …), this package allows users to define arbitrary character
or numeric filters using the <code>VariableFilter</code>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mito_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-filtering.html">VariableFilter</a></span><span class="op">(</span>field <span class="op">=</span> <span class="st">"markers"</span>,</span>
<span>                              value <span class="op">=</span> <span class="st">"Mitochondrion"</span>,</span>
<span>                              condition <span class="op">=</span> <span class="st">"=="</span><span class="op">)</span></span>
<span><span class="va">mito_filter</span></span></code></pre></div>
<pre><code><span><span class="co">## class: CharacterVariableFilter </span></span>
<span><span class="co">## condition: == </span></span>
<span><span class="co">## value: Mitochondrion</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qval_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-filtering.html">VariableFilter</a></span><span class="op">(</span>field <span class="op">=</span> <span class="st">"qValue"</span>,</span>
<span>                              value <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>                              condition <span class="op">=</span> <span class="st">"&lt;="</span><span class="op">)</span></span>
<span><span class="va">qval_filter</span></span></code></pre></div>
<pre><code><span><span class="co">## class: NumericVariableFilter </span></span>
<span><span class="co">## condition: &lt;= </span></span>
<span><span class="co">## value: 0.001</span></span></code></pre>
<p>These filter can then readily be applied to all assays’ row data
slots. The <code>mito_filter</code> will return all PSMs, peptides and
proteins that were annotated as localising to the mitochondrion.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="va">mito_filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'markers' found in 3 out of 3 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 167 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 162 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 113 rows and 10 columns</span></span></code></pre>
<p>The <code>qval_filter</code>, on the other hand, will only return a
subset of PSMs, because the <code>qValue</code> variable is only present
in the <code>psms</code> assays. The q-values are only relevant to PSMs
and that variable was dropped from the other assays.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="va">qval_filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'qValue' found in 1 out of 3 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## No filter applied to the following assay(s) because one or more</span></span>
<span><span class="co">## filtering variables are missing in the rowData: peptides, proteins. You</span></span>
<span><span class="co">## can control whether to remove or keep the features using the 'keep'</span></span>
<span><span class="co">## argument (see '?filterFeature').</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 2466 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 0 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 0 rows and 10 columns</span></span></code></pre>
<p>The same filters can be created using the forumla interface:</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="op">~</span> <span class="va">markers</span> <span class="op">==</span> <span class="st">"Mitochondrion"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'markers' found in 3 out of 3 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 167 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 162 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 113 rows and 10 columns</span></span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="op">~</span> <span class="va">qValue</span> <span class="op">&lt;=</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'qValue' found in 1 out of 3 assay(s).</span></span></code></pre>
<pre><code><span><span class="co">## No filter applied to the following assay(s) because one or more</span></span>
<span><span class="co">## filtering variables are missing in the rowData: peptides, proteins. You</span></span>
<span><span class="co">## can control whether to remove or keep the features using the 'keep'</span></span>
<span><span class="co">## argument (see '?filterFeature').</span></span></code></pre>
<pre><code><span><span class="co">## An instance of class QFeatures (type: bulk) with 3 sets:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  [1] psms: SummarizedExperiment with 2466 rows and 10 columns </span></span>
<span><span class="co">##  [2] peptides: SummarizedExperiment with 0 rows and 10 columns </span></span>
<span><span class="co">##  [3] proteins: SummarizedExperiment with 0 rows and 10 columns</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ggplot2_4.0.0               QFeatures_1.19.4           </span></span>
<span><span class="co">##  [3] MultiAssayExperiment_1.35.9 SummarizedExperiment_1.39.2</span></span>
<span><span class="co">##  [5] Biobase_2.69.1              GenomicRanges_1.61.5       </span></span>
<span><span class="co">##  [7] Seqinfo_0.99.2              IRanges_2.43.5             </span></span>
<span><span class="co">##  [9] S4Vectors_0.47.4            BiocGenerics_0.55.4        </span></span>
<span><span class="co">## [11] generics_0.1.4              MatrixGenerics_1.21.0      </span></span>
<span><span class="co">## [13] matrixStats_1.5.0           BiocStyle_2.37.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] gtable_0.3.6            xfun_0.53               bslib_0.9.0            </span></span>
<span><span class="co">##  [4] htmlwidgets_1.6.4       lattice_0.22-7          vctrs_0.6.5            </span></span>
<span><span class="co">##  [7] tools_4.5.1             tibble_3.3.0            cluster_2.1.8.1        </span></span>
<span><span class="co">## [10] pkgconfig_2.0.3         BiocBaseUtils_1.11.2    Matrix_1.7-4           </span></span>
<span><span class="co">## [13] RColorBrewer_1.1-3      S7_0.2.0                desc_1.4.3             </span></span>
<span><span class="co">## [16] lifecycle_1.0.4         farver_2.1.2            compiler_4.5.1         </span></span>
<span><span class="co">## [19] stringr_1.5.2           textshaping_1.0.4       clue_0.3-66            </span></span>
<span><span class="co">## [22] htmltools_0.5.8.1       sass_0.4.10             yaml_2.3.10            </span></span>
<span><span class="co">## [25] lazyeval_0.2.2          pkgdown_2.1.3.9000      pillar_1.11.1          </span></span>
<span><span class="co">## [28] crayon_1.5.3            jquerylib_0.1.4         tidyr_1.3.1            </span></span>
<span><span class="co">## [31] MASS_7.3-65             DelayedArray_0.35.3     cachem_1.1.0           </span></span>
<span><span class="co">## [34] abind_1.4-8             tidyselect_1.2.1        digest_0.6.37          </span></span>
<span><span class="co">## [37] stringi_1.8.7           purrr_1.1.0             dplyr_1.1.4            </span></span>
<span><span class="co">## [40] reshape2_1.4.4          bookdown_0.45           labeling_0.4.3         </span></span>
<span><span class="co">## [43] fastmap_1.2.0           grid_4.5.1              cli_3.6.5              </span></span>
<span><span class="co">## [46] SparseArray_1.9.1       magrittr_2.0.4          S4Arrays_1.9.1         </span></span>
<span><span class="co">## [49] withr_3.0.2             scales_1.4.0            rmarkdown_2.30         </span></span>
<span><span class="co">## [52] XVector_0.49.1          igraph_2.2.0            ragg_1.5.0             </span></span>
<span><span class="co">## [55] evaluate_1.0.5          knitr_1.50              rlang_1.1.6            </span></span>
<span><span class="co">## [58] Rcpp_1.1.0              glue_1.8.0              BiocManager_1.30.26    </span></span>
<span><span class="co">## [61] jsonlite_2.0.0          AnnotationFilter_1.33.0 R6_2.6.1               </span></span>
<span><span class="co">## [64] plyr_1.8.9              systemfonts_1.3.1       fs_1.6.6               </span></span>
<span><span class="co">## [67] ProtGenerics_1.41.0     MsCoreUtils_1.21.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>This vignette is distributed under a <a href="https://creativecommons.org/licenses/by-sa/2.0/" class="external-link">CC BY-SA
license</a> license.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Christoforou:2016" class="csl-entry">
Christoforou, Andy, Claire M Mulvey, Lisa M Breckels, et al. 2016.
<span>“A Draft Map of the Mouse Pluripotent Stem Cell Spatial
Proteome.”</span> <em>Nat Commun</em> 7: 8992. <a href="https://doi.org/10.1038/ncomms9992" class="external-link">https://doi.org/10.1038/ncomms9992</a>.
</div>
<div id="ref-SE" class="csl-entry">
Morgan, Martin, Valerie Obenchain, Jim Hester, and Hervé Pagès. 2019.
<em><span>SummarizedExperiment</span>: <span>SummarizedExperiment</span>
Container</em>.
</div>
<div id="ref-AnnotationFilter" class="csl-entry">
Morgan, Martin, and Johannes Rainer. 2019.
<em><span>AnnotationFilter</span>: Facilities for Filtering
<span>Bioconductor</span> Annotation Resources</em>. <a href="https://github.com/Bioconductor/AnnotationFilter" class="external-link">https://github.com/Bioconductor/AnnotationFilter</a>.
</div>
<div id="ref-Ramos:2017" class="csl-entry">
Ramos, Marcel, Lucas Schiffer, Angela Re, et al. 2017. <span>“Software
for the Integration of Multi-Omics Experiments in
<span>Bioconductor</span>.”</span> <em>Cancer Research</em> 77(21);
e39-42.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Laurent Gatto, Christophe Vanderaa.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
