<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data visualization from a QFeatures object • QFeatures</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data visualization from a QFeatures object">
<meta property="og:description" content="QFeatures">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">QFeatures</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/QFeatures.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Processing.html">Processing quantitative proteomics data with QFeatures</a>
    </li>
    <li>
      <a href="../articles/Visualization.html">Data visualization from a QFeatures object</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RforMassSpectrometry/QFeatures/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data visualization from a QFeatures object</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/RforMassSpectrometry/QFeatures/blob/HEAD/vignettes/Visualization.Rmd" class="external-link"><code>vignettes/Visualization.Rmd</code></a></small>
      <div class="hidden name"><code>Visualization.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to visualize quantitative mass
      spectrometry data contained in a QFeatures object.</p>
    </div>
    
<div class="section level2">
<h2 id="preparing-the-data">Preparing the data<a class="anchor" aria-label="anchor" href="#preparing-the-data"></a>
</h2>
<p>To demonstrate the data visualization of a <code>QFeatures</code>, we
first perform a quick processing of the <code>hlpsms</code> example
data. We load the data and read it as a <code>QFeautres</code> object.
See the processing <a href="https://rformassspectrometry.github.io/QFeatures/articles/Processing.html" class="external-link">vignette</a>
for more details about data processing with <code>QFeatures</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/RforMassSpectrometry/QFeatures" class="external-link">"QFeatures"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hlpsms</span><span class="op">)</span>
<span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readQFeatures.html">readQFeatures</a></span><span class="op">(</span><span class="va">hlpsms</span>, ecol <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, name <span class="op">=</span> <span class="st">"psms"</span><span class="op">)</span></code></pre></div>
<p>We then aggregate the psms to peptides, and the peptodes to
proteins.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="st">"psms"</span>, <span class="st">"Sequence"</span>, name <span class="op">=</span> <span class="st">"peptides"</span>, fun <span class="op">=</span> <span class="va">colMeans</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Your row data contain missing values. Please read the relevant</span>
<span class="co">## section(s) in the aggregateFeatures manual page regarding the effects</span>
<span class="co">## of missing values on data aggregation.</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">hl</span>, <span class="st">"peptides"</span>, <span class="st">"ProteinGroupAccessions"</span>, name <span class="op">=</span> <span class="st">"proteins"</span>, fun <span class="op">=</span> <span class="va">colMeans</span><span class="op">)</span></code></pre></div>
<p>We also add the TMT tags that were used to multiplex the samples. The
data is added to the <code>colData</code> of the <code>QFeatures</code>
object and will allow us to demonstrate how to plot data from the
<code>colData</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hl</span><span class="op">$</span><span class="va">tag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"126"</span>, <span class="st">"127N"</span>, <span class="st">"127C"</span>, <span class="st">"128N"</span>, <span class="st">"128C"</span>, <span class="st">"129N"</span>, <span class="st">"129C"</span>,
            <span class="st">"130N"</span>, <span class="st">"130C"</span>, <span class="st">"131"</span><span class="op">)</span></code></pre></div>
<p>The dataset is now ready for data exploration.</p>
</div>
<div class="section level2">
<h2 id="exploring-the-qfeatures-hierarchy">Exploring the <code>QFeatures</code> hierarchy<a class="anchor" aria-label="anchor" href="#exploring-the-qfeatures-hierarchy"></a>
</h2>
<p><code>QFeatures</code> objects can contain several assays as the data
goes through the processing workflow. The <code>plot</code> function
provides an overview of all the assays present in the dataset, showing
also the hierarchical relationships between the assays as determined by
the <code>AssayLinks</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/plot-1.png" width="700"></p>
<p>This plot is rather simple with only three assays, but some
processing workflows may involve more steps. The <code>feat3</code>
example data illustrates the different possible relationships: one
parent to one child, multiple parents to one child and one parent to
multiple children.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"feat3"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">feat3</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/plot2-1.png" width="400px"></p>
<p>Note that some datasets may contain many assays, for instance because
the MS experiment consists of hundreds of batches. This can lead to an
overcrowded plot. Therefore, you can also explore this hierarchy of
assays through an interactive plot, supported by the <code>plotly</code>
package (<span class="citation">Sievert (2020)</span>). You can use the
viewer panel to zoom in and out and navigate across the tree(s).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">hl</span>, interactive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-data-exploration">Basic data exploration<a class="anchor" aria-label="anchor" href="#basic-data-exploration"></a>
</h2>
<p>The quantitative data is retrieved using <code>assay()</code>, the
feature metadata is retrieved using <code>rowData()</code> on the assay
of interest, and the sample metadata is retrieved using
<code>colData()</code>. Once retrieved, the data can be supplied to the
base R data exploration tools. Here are some examples:</p>
<ul>
<li>Plot the intensities for the first protein. These data are available
from the <code>proteins</code> assay.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">hl</span>, <span class="st">"proteins"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>, <span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/plot_assay-1.png" width="700"></p>
<ul>
<li>Get the distribution of the number of peptides that were aggregated
per protein. These data are available in the column <code>.n</code> from
the protein <code>rowData</code>.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">.n</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/hist_rowData-1.png" width="700"></p>
<ul>
<li>Get the count table of the different tags used for labeling the
samples. These data are available in the column <code>tag</code> from
the <code>colData</code>.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hl</span><span class="op">$</span><span class="va">tag</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  126 127C 127N 128C 128N 129C 129N 130C 130N  131 </span>
<span class="co">##    1    1    1    1    1    1    1    1    1    1</span></code></pre>
</div>
<div class="section level2">
<h2 id="using-ggplot2">Using <code>ggplot2</code><a class="anchor" aria-label="anchor" href="#using-ggplot2"></a>
</h2>
<p><code>ggplot2</code> is a powerful tool for data visualization in
<code>R</code> and is part of the <code>tidyverse</code> package
ecosystem (<span class="citation">Wickham et al. (2019)</span>). It
produces elegant and publication-ready plots in a few lines of code.
<code>ggplot2</code> can be used to explore <code>QFeatures</code>
object, similarly to the base functions shown above. Note that
<code>ggplot2</code> expects <code>data.frame</code> or
<code>tibble</code> objects whereas the quantitative data in
<code>QFeatures</code> are encoded as <code>matrix</code> (or
matrix-like objects, see <code>?SummarizedExperiment</code>) and the
<code>rowData</code> and <code>colData</code> are encoded as
<code>DataFrame</code>. This is easily circumvented by converting those
objects to <code>data.frame</code>s or <code>tibble</code>s. See here
how we reproduce the plot above using <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.n</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/ggplot_rowData-1.png" width="700"></p>
<p>We refer the reader to the <code>ggplot2</code> <a href="https://ggplot2.tidyverse.org/" class="external-link">package website</a> for more
information about the wide variety of functions that the package offers
and for tutorials and cheatsheets.</p>
<p>Another useful package for quantitative data exploration is
<code>ComplexHeatmap</code> (<span class="citation">Gu, Eils, and
Schlesner (2016)</span>). It is part of the Bioconductor project (<span class="citation">Gentleman et al. (2004)</span>) and facilitates
visualization of matrix objects as heatmap. See here an example where we
plot the protein data.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span>matrix <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">hl</span>, <span class="st">"proteins"</span><span class="op">)</span>,
        show_row_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/ComplexHeatmap-1.png" width="700"></p>
<p><code>ComplexHeatmap</code> also allows to add row and/or column
annotations. Let’s add the predicted protein location as row
annotation.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ha</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>markers <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">hl</span><span class="op">)</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">markers</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span>matrix <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">hl</span>, <span class="st">"proteins"</span><span class="op">)</span>,
        show_row_names <span class="op">=</span> <span class="cn">FALSE</span>,
        left_annotation <span class="op">=</span> <span class="va">ha</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/ComplexHeatmap_annotations-1.png" width="700"></p>
<p>More advanced usage of <code>ComplexHeatmap</code> is described in
the package reference <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/" class="external-link">book</a>.</p>
</div>
<div class="section level2">
<h2 id="advanced-data-exploration">Advanced data exploration<a class="anchor" aria-label="anchor" href="#advanced-data-exploration"></a>
</h2>
<p>In this section, we show how to combine in a single table different
pieces of information available in a <code>QFeatures</code> object, that
are quantitation data, feature metadata and sample metadata. The
<code>QFeatures</code> package provides the <code>longFormat</code>
function that converts a <code>QFeatures</code> object into a long
table. Long tables are very useful when using <code>ggplot2</code> for
data visualization. For instance, suppose we want to visualize the
distribution of protein quantitation (present in the
<code>proteins</code> assay) with respect to the different acquisition
tags (present in the <code>colData</code>) for each predicted cell
location separately (present in the <code>rowData</code> of the assays).
Furthermore, we link the quantitation values coming from the same
protein using lines. This can all be plotted at once in a few lines of
code.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QFeatures-class.html">longFormat</a></span><span class="op">(</span><span class="va">hl</span><span class="op">[</span>, , <span class="st">"proteins"</span><span class="op">]</span>,
                 rowvars <span class="op">=</span> <span class="st">"markers"</span>,
                 colvars <span class="op">=</span> <span class="st">"tag"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></code></pre>
<pre><code><span class="co">## harmonizing input:</span>
<span class="co">##   removing 20 sampleMap rows not in names(experiments)</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">lf</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tag</span>,
        y <span class="op">=</span> <span class="va">value</span>,
        group <span class="op">=</span> <span class="va">rowname</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">markers</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="Visualization_files/figure-html/longFormat-1.png" width="864"></p>
<p><code>longFormat</code> allows to retrieve and combine all available
data from a <code>Qfeatures</code> object. We here demonstrate the ease
to combine different pieces that could highlight sample specific and/or
feature specific effects on data quantitation.</p>
</div>
<div class="section level2">
<h2 id="interactive-data-exploration">Interactive data exploration<a class="anchor" aria-label="anchor" href="#interactive-data-exploration"></a>
</h2>
<p>Finally, a simply <code>shiny</code> app allows to explore and
visualise the respective assays of a <code>QFeatures</code> object.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/display.html">display</a></span><span class="op">(</span><span class="va">hl</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figs/display_hmap.png" alt="`QFeatures` interactive interface: heatmap of the peptide assay data." width="100%"><p class="caption">
<code>QFeatures</code> interactive interface: heatmap of the peptide
assay data.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figs/display_assay.png" alt="`QFeatures` interactive interface: quantitative peptide assay data." width="100%"><p class="caption">
<code>QFeatures</code> interactive interface: quantitative peptide assay
data.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figs/display_rowdata.png" alt="`QFeatures` interactive interface: peptide assay row data" width="100%"><p class="caption">
<code>QFeatures</code> interactive interface: peptide assay row data
</p>
</div>
<p>A dropdown menu in the side bar allows the user to select an assay of
interest, which can then be visualised as a heatmap (figure
@ref(fig:heatmapdisplay)), as a quantitative table (figure
@ref(fig:assaydisplay)) or a row data table (figure
@ref(fig:rowdatadisplay)).</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span class="co">## R version 4.2.0 (2022-04-22)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">## [8] methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] ComplexHeatmap_2.13.0       ggplot2_3.3.6              </span>
<span class="co">##  [3] QFeatures_1.7.1             MultiAssayExperiment_1.23.2</span>
<span class="co">##  [5] SummarizedExperiment_1.27.1 Biobase_2.57.1             </span>
<span class="co">##  [7] GenomicRanges_1.49.0        GenomeInfoDb_1.33.3        </span>
<span class="co">##  [9] IRanges_2.31.0              S4Vectors_0.35.1           </span>
<span class="co">## [11] BiocGenerics_0.43.0         MatrixGenerics_1.9.1       </span>
<span class="co">## [13] matrixStats_0.62.0          BiocStyle_2.25.0           </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] ProtGenerics_1.27.2     bitops_1.0-7            fs_1.5.2               </span>
<span class="co">##  [4] doParallel_1.0.17       RColorBrewer_1.1-3      rprojroot_2.0.3        </span>
<span class="co">##  [7] tools_4.2.0             bslib_0.3.1             utf8_1.2.2             </span>
<span class="co">## [10] R6_2.5.1                lazyeval_0.2.2          colorspace_2.0-3       </span>
<span class="co">## [13] GetoptLong_1.0.5        withr_2.5.0             tidyselect_1.1.2       </span>
<span class="co">## [16] compiler_4.2.0          textshaping_0.3.6       cli_3.3.0              </span>
<span class="co">## [19] desc_1.4.1              DelayedArray_0.23.0     labeling_0.4.2         </span>
<span class="co">## [22] bookdown_0.27           sass_0.4.1              scales_1.2.0           </span>
<span class="co">## [25] pkgdown_2.0.6.9000      systemfonts_1.0.4       stringr_1.4.0          </span>
<span class="co">## [28] digest_0.6.29           rmarkdown_2.14          XVector_0.37.0         </span>
<span class="co">## [31] pkgconfig_2.0.3         htmltools_0.5.2         fastmap_1.1.0          </span>
<span class="co">## [34] highr_0.9               rlang_1.0.3             GlobalOptions_0.1.2    </span>
<span class="co">## [37] shape_1.4.6             jquerylib_0.1.4         farver_2.1.0           </span>
<span class="co">## [40] generics_0.1.2          jsonlite_1.8.0          dplyr_1.0.9            </span>
<span class="co">## [43] RCurl_1.98-1.7          magrittr_2.0.3          GenomeInfoDbData_1.2.8 </span>
<span class="co">## [46] Matrix_1.4-1            Rcpp_1.0.8.3            munsell_0.5.0          </span>
<span class="co">## [49] fansi_1.0.3             MsCoreUtils_1.9.0       lifecycle_1.0.1        </span>
<span class="co">## [52] stringi_1.7.6           yaml_2.3.5              MASS_7.3-57            </span>
<span class="co">## [55] zlibbioc_1.43.0         parallel_4.2.0          crayon_1.5.1           </span>
<span class="co">## [58] lattice_0.20-45         circlize_0.4.15         magick_2.7.3           </span>
<span class="co">## [61] knitr_1.39              pillar_1.7.0            igraph_1.3.2           </span>
<span class="co">## [64] rjson_0.2.21            codetools_0.2-18        glue_1.6.2             </span>
<span class="co">## [67] evaluate_0.15           BiocManager_1.30.18     vctrs_0.4.1            </span>
<span class="co">## [70] png_0.1-7               foreach_1.5.2           gtable_0.3.0           </span>
<span class="co">## [73] purrr_0.3.4             clue_0.3-61             cachem_1.0.6           </span>
<span class="co">## [76] xfun_0.31               AnnotationFilter_1.21.0 ragg_1.2.2             </span>
<span class="co">## [79] tibble_3.1.7            iterators_1.0.14        memoise_2.0.1          </span>
<span class="co">## [82] cluster_2.1.3           ellipsis_0.3.2</span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Gentleman:2004" class="csl-entry">
Gentleman, Robert C., Vincent J. Carey, Douglas M. Bates, Ben Bolstad,
Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004.
<span>“Bioconductor: Open Software Development for Computational Biology
and Bioinformatics.”</span> <em>Genome Biol</em> 5 (10): –80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80" class="external-link">https://doi.org/10.1186/gb-2004-5-10-r80</a>.
</div>
<div id="ref-Gu2016-ej" class="csl-entry">
Gu, Zuguang, Roland Eils, and Matthias Schlesner. 2016. <span>“Complex
Heatmaps Reveal Patterns and Correlations in Multidimensional Genomic
Data.”</span> <em>Bioinformatics</em> 32 (18): 2847–49.
</div>
<div id="ref-Sievert2020" class="csl-entry">
Sievert, Carson. 2020. <em>Interactive Web-Based Data Visualization with
r, Plotly, and Shiny</em>. Chapman; Hall/CRC. <a href="https://plotly-r.com" class="external-link">https://plotly-r.com</a>.
</div>
<div id="ref-Wickham2019-fz" class="csl-entry">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy
McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome
to the Tidyverse.”</span> <em>J. Open Source Softw.</em> 4 (43): 1686.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto, Christophe Vanderaa.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
